Code/Framework/AzToolsFramework/Tests/PythonLoaderTests.cpp

```diff
@@ -0,0 +1,126 @@
+/*
+ * Copyright (c) Contributors to the Open 3D Engine Project.
+ * For complete copyright and license terms please see the LICENSE at the root of this distribution.
+ *
+ * SPDX-License-Identifier: Apache-2.0 OR MIT
+ *
+ */
+
+#include <AzCore/Component/Entity.h>
+#include <AzCore/RTTI/BehaviorContext.h>
+#include <AzCore/Serialization/SerializeContext.h>
+#include <AzCore/std/containers/vector.h>
+#include <AzCore/std/sort.h>
+#include <AzTest/AzTest.h>
+#include <AzTest/Utils.h>
+#include <AzToolsFramework/Application/ToolsApplication.h>
+#include <AzToolsFramework/UnitTest/AzToolsFrameworkTestHelpers.h>
+#include <AzToolsFramework/Entity/EditorEntityContextComponent.h>
+#include <AzCore/UnitTest/TestTypes.h>
+#include <AzFramework/IO/LocalFileIO.h>
+#include <AzToolsFramework/API/PythonLoader.h>
+
+namespace UnitTest
+{
+    class AzToolsFrameworkPythonLoaderFixture
+        : public ToolsApplicationFixture<>
+    {
+    protected:
+        AZ::Test::ScopedAutoTempDirectory m_tempDirectory;
+
+        static constexpr const char* s_testEnginePath = "O3de_path";
+        static constexpr const char* s_testEnginePathHashId = "1af80774";
+        static constexpr const char* s_test3rdPartySubPath = ".o3de/3rdParty";
+    };
+
+    TEST_F(AzToolsFrameworkPythonLoaderFixture, TestGetPythonVenvPath_Valid)
+    {
+        auto test3rdPartyPath = m_tempDirectory.GetDirectoryAsFixedMaxPath() / s_test3rdPartySubPath;
+        auto testVenvPath = test3rdPartyPath / "venv";
+
+        AZStd::string engineRoot{ s_testEnginePath };
+
+        AZ::IO::FileIOBase::GetInstance()->CreatePath(test3rdPartyPath.String().c_str());
+
+        auto result = AzToolsFramework::EmbeddedPython::PythonLoader::GetPythonVenvPath(test3rdPartyPath, engineRoot);
+
+        AZ::IO::FixedMaxPath expectedPath = testVenvPath / s_testEnginePathHashId;
+
+        EXPECT_TRUE(result == expectedPath);
+    }
+
+    TEST_F(AzToolsFrameworkPythonLoaderFixture, TestGetPythonVenvExecutablePath_Valid)
+    {
+        // Prepare the test venv pyvenv.cfg file in the expected location
+        AZ::IO::FixedMaxPath tempVenvRelativePath = AZ::IO::FixedMaxPath(s_test3rdPartySubPath) / "venv/" / s_testEnginePathHashId;
+        AZ::IO::FixedMaxPath tempVenvFullPath = m_tempDirectory.GetDirectoryAsFixedMaxPath() / tempVenvRelativePath;
+        AZ::IO::FileIOBase::GetInstance()->CreatePath(tempVenvFullPath.String().c_str());
```

Comment:
The ToolApplicationFixture can  be avoided, by not relying on the AZ::IO::FileIOBase.
That can be done by using `AZ::IO::SystemFile::CreateDir` to create directories underneath the temporary directory, without needing to create  an `AzFramework::Application` in order to initialize an `AZ::IO::LocalFileIO`

Reply:
I replaced with SystemFile::CreateDir, but I still need `AZ::IO::FileIOBase::GetDirectInstance` because for `PythonLoader::ReadPythonEggLinkPaths`, it relies on `AZ::IO::FileIOBase::GetDirectInstance()->FindFiles`. The SystemFile::FindFiles does not recursively search like the `LocalFileIO` version does, and converting to it will unnecessarily make this function more complicated than it needs to.